python/lsst/pipe/base/activator.py

```diff
@@ -0,0 +1,449 @@
+"""
+activator: This module contains the activator class which are used
+to execute (Super)Task in different context/environments
+"""
+from __future__ import absolute_import, division, print_function
+#
+# LSST Data Management System
+# Copyright 2008-2013 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import sys
+import inspect
+import pkgutil
+import pyclbr
+import contextlib
+from .argumentParser import ArgumentParser
+import argparse as argp
+
+import importlib
+
+
+__all__ = ["Activator", "CmdLineActivator"]
+
+# This is the temporary place for all the packages with (Super)tasks inside
+TASK_PACKAGES = {'lsst.pipe.base.examples': None, 'lsst.pipe.tasks': None}
+
+for pkg in TASK_PACKAGES.keys():
+    # This get the list of modules inside package, from the __init__.py inside package
+    TASK_PACKAGES[pkg] = importlib.import_module(pkg)
+
+
+class ActivatorParser(argp.ArgumentParser):
+    """
+    Small ActivatorParser handler to print error message to add an error method
+    """
+
+    def error(self, message):
+        """
+        Writes outs the message error in the argument parser
+        """
+        sys.stderr.write('error: %s\n' % message)
+        self.print_help()
+        sys.exit(2)
+
+
+@contextlib.contextmanager
+def profile(filename, log=None):
+    """!Context manager for profiling with cProfile
+
+    @param filename     filename to which to write profile (profiling disabled if None or empty)
+    @param log          log object for logging the profile operations
+
+    If profiling is enabled, the context manager returns the cProfile.Profile object (otherwise
+    it returns None), which allows additional control over profiling.  You can obtain this using
+    the "as" clause, e.g.:
```

Comment:
I don't see any test for "if profiling not enabled then return None". Is this accurate? (And why return None instead of a disabled Profile()?)